src/guardian/guardian.service.ts

```diff
@@ -280,10 +297,34 @@ export class GuardianService implements OnModuleInit {
     return intersections;
   }
 
+  /**
+   * Excludes invalid deposits and deposits with Lido WC from intersections
+   * @param intersections - list of deposits with keys that were deposited earlier
+   * @param blockData - collected data from the current block
+   */
+  public async excludeEligibleIntersections(
+    intersections: VerifiedDepositEvent[],
+    blockData: BlockData,
+  ): Promise<VerifiedDepositEvent[]> {
+    // Exclude deposits with invalid signature over the deposit data
+    const validIntersections = intersections.filter(({ valid }) => valid);
+    if (!validIntersections.length) return [];
+
+    // Exclude deposits with Lido withdrawal credentials
+    const { blockHash } = blockData;
+    const lidoWC = await this.lidoService.getWithdrawalCredentials({
+      blockHash,
+    });
+    const attackIntersections = validIntersections.filter(
```

Comment:
<img width="541" alt="Screenshot 2022-07-14 at 15 29 52" src="https://user-images.githubusercontent.com/102730938/178982610-ef95a07d-4a51-4536-87ef-a886a73f9987.png">

My ide complains about a variable. So, It'd be great to set up a project with common linter rules.